### PR TITLE
Adding date range time example to docs.

### PR DESCRIPTION
**Pull request by stephenworsley**
Adding the example mentioned in #3235 to the docs.

### CODE REVIEW

**Comment by bjlittle** [comment]
@stephenworsley Awesome, thanks.

You've got some Travis CI failures. Could you re-base your branch against `master` and push this to your branch?

Hopefully this should resolve these issues.

**Comment by bjlittle** [comment]
@stephenworsley Thanks :+1: 

I'll just take a peek at the rendered documentation changes, and take it from there...

**Comment by bjlittle** [comment]
@stephenworsley Nice. Good job :smile: 

Since we're in this space, could I cheekily ask for an additional change, just to bed in the example of using a date range in a constraint **and** the use of `iris.time.PartialDateTime` objects.

Just after the section:
```python
>>> import datetime
>>> from iris.time import PartialDateTime
>>> dt = datetime.datetime(2011, 3, 7)
>>> print(dt > PartialDateTime(year=2010, month=6))
True
>>> print(dt > PartialDateTime(month=6))
False
>>>
```
```These two facilities can be combined to provide straightforward calendar-based time selections when loading or extracting data.```
Could we bring it all together with a further example along the lines of:
```python
>>> from iris.time import PartialDateTime
>>> pdt1 = PartialDateTime(year=2007, month=4, day=16)
>>> pdt2 = PartialDateTime(year=2007, month=4, day=30)
>>> daterange = iris.Constraint(time=lambda cell: d1<= cell.point <= d2)
>>> data_subset = long_ts.extract(daterange)
>>> print(data_subset.coord('time'))
```
which builds on your new example, but using `PartialDateTime` objects instead... what do you think?

Of couse, add some suitable introductory text, as you see fit.

It also means that we might need to slightly rephrase the ```The previous constraint example can now be written as:``` sentence, as it's slightly confusing as to which constraint it's referring to - clearly the `hour_11` constraint.

**Comment by bjlittle** [comment]
@stephenworsley Also if you make your next commit message "closes #3235", this mean that once this PR is merge, GitHub will automatically close the associated issue, see https://help.github.com/en/articles/closing-issues-using-keywords

**Comment by stephenworsley** [comment]
@bjlittle I believe there is already an example doing more or less the same thing with the `st_swithuns_daterange`, though it might make sense to modify the dates in the new example to match up. I'll see what I can do about clearing up any ambiguity in the 'previous' references.

**Comment by bjlittle** [comment]
@stephenworsley Yeah, I think that it's the overall flow of that section is just not quite right... there are a mixture of dates, constraints and examples and for first time reader, I can only think that it's harder work that it should be.

It would be awesome for you could manage to smooth out the wrinkles and yet get across all the relevant points of time constrains.

See how you go, and we can always talk about it in person tomorrow if it's easier :smile:

File: docs/iris/src/userguide/loading_iris_cubes.rst [hunk 1]
```diff
@@ -314,21 +315,53 @@ for many years:
...
+A more complex example might require selecting points over an annually repeating
+date range. We can select points within a certain part of the year, in this case
+between the 15th of July through to the 25th of August. By making use of
+PartialDateTime this becomes simple:
 
 .. doctest:: timeseries_range
 
     >>> st_swithuns_daterange = iris.Constraint(
-    ...     time=lambda cell: PartialDateTime(month=7, day=15) < cell < PartialDateTime(month=8, day=25))
+    ...     time=lambda cell: PartialDateTime(month=7, day=15) <= cell < PartialDateTime(month=8, day=25))
```
Comment (lbdreyer) [inline docs/iris/src/userguide/loading_iris_cubes.rst]: Why this change?

File: docs/iris/src/userguide/loading_iris_cubes.rst [hunk 2]
```diff
@@ -314,21 +315,53 @@ for many years:
...
+    >>> st_swithuns_daterange_07 = iris.Constraint(
+    ...     time=lambda cell: d1 <= cell.point < d2)
+    >>> within_st_swithuns_07 = long_ts.extract(st_swithuns_daterange_07)
+    >>> print(within_st_swithuns_07.coord('time'))
+    DimCoord([2007-07-16 00:00:00, 2007-07-23 00:00:00, 2007-07-30 00:00:00,
+              2007-08-06 00:00:00, 2007-08-13 00:00:00, 2007-08-20 00:00:00],
+             standard_name='time', calendar='gregorian')
+
+Alternatively, we may do the same thing with :class:`iris.time.PartialDateTime`
+objects.
```
Comment (lbdreyer) [inline docs/iris/src/userguide/loading_iris_cubes.rst]: > Alternatively, we may do the same thing with ...

This phrasing feels a bit too informal.
Could you rephrase to be something like:
> Alternatively, we may rewrite this using PartialDateTime objects ...

File: docs/iris/src/userguide/loading_iris_cubes.rst (hunk 1 above)
Comment (stephenworsley) [reply docs/iris/src/userguide/loading_iris_cubes.rst]: So there's two reasons for this. The first is that I wanted to make it more obvious that PartialDateTime range was the same as the range defined by datetetime objects. The datetime objects measure time from the start of the day, so for them it is necessary to specify the first date in the range and the first date outside of the range. With PartialDateTime, this is not necessary but I made the change so that the dates matched up.
The other reason is that this example appears to be a reference to a proverb about St Swithun's day https://en.wikipedia.org/wiki/Swithun. I decided that an object called `st_swithuns_daterange` could reasonably be expected to contain St Swithun's day itself (this does mean that there are now 41 days in the daterange, though this still fits with some readings of the proverb).
